fetch-downloads.py

```diff
@@ -190,8 +197,15 @@
         latest_grouped = defaultdict(lambda: ["", ""])
         versions = await get_listing(type)
 
+        limit = limits.get(type)
+
         # Regroup the versions based on the method
         for version, data in versions.items():
+            if limit is not None:
+                if limit == 0:
+                    break
+                limit -= 1
```

Comment:
this all seems a bit hacky. given the keys are in an "arbitrary" order (even if that's not really the case in 3.6+), this seems like the wrong place to limit the number of versions

Reply:
I disagree ;) But this is a closed comment that is not helpful to me. If you have a better suggestion, please let me know.

Do know, this whole script is 3.6+ only, as it fully depends on the fact dicts are in order (that was already the case, that is still the case). So adding additional "sorted" etc only clutters the code. So I am a bit at a loss here where this could be improved. This is clear, and to the point. Correct, and with the least amount of code. Not sure what you would like to see more.

As for the wrong place ... this to me sounds like exactly the right place? Not sure where else you would want to do it?

Again, I am just not sure what to do with this comment :) Sorry :) I do appreciate the feedback! Just wish I could act on it :)

Reply:
I don't like the use of `limit` has a loop counter, mostly. Perhaps something like `...items()[:limit]` would be nicer?

Reply:
`TypeError: 'dict_items' object is not subscriptable`. To solve that I have to make it a list already (instead of an iter), which would be very expensive on the memory. Not sure that is a bad thing, but that felt worse :D